src/include/common/data_chunk/data_chunk_state.h

```diff
@@ -27,21 +28,26 @@ class DataChunkState {
         originalSize = size;
         selVector->selectedSize = size;
     }
-    inline bool isFlat() const { return currIdx != -1; }
-    inline void setToUnflat() { currIdx = -1; }
-    inline uint64_t getNumSelectedValues() const { return isFlat() ? 1 : selVector->selectedSize; }
+    inline void setOriginalSize(uint64_t size) { originalSize = size; }
+    inline uint64_t getOriginalSize() { return originalSize; }
+    inline bool isFlat() const { return factorizationState == FactorizationStateType::FLAT; }
+    inline void setToFlat() { factorizationState = FactorizationStateType::FLAT; }
+    inline void setToUnflat() { factorizationState = FactorizationStateType::UNFLAT; }
+
+    inline uint64_t getNumSelectedValues() const { return selVector->selectedSize; }
 
 public:
-    // The currIdx is >= 0 when vectors are flattened and -1 if the vectors are unflat.
-    int64_t currIdx;
+    std::shared_ptr<SelectionVector> selVector;
```

Comment:
Also move this field `selVector` to private?

Reply:
This involves a lot of change so should be done separately.